engine/src/main/java/org/terasology/world/chunks/blockdata/ExtraDataSystem.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2018 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.world.chunks.blockdata;
+
+import org.terasology.module.sandbox.API;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation for classes which register extra per-block data fields.
+ * Methods for this purpose must also be annotated with {@link org.terasology.world.chunks.blockdata.RegisterExtraData}.
+ */
+@API
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+public @interface ExtraDataSystem {
```

Comment:
I thought this made sense by analogy to `ComponentSystem`. `ComponentSystem`s are systems for dealing with component events, and `ExtraDataSystem`s are systems for dealing with extra-data field registration, the common theme being that the engine finds the class and calls its methods. It's possible to have something annotated with `ExtraDataSystem` that isn't a `ComponentSystem`, so an annotation name that only makes sense in that context wouldn't be as good.

I think I might just have  slightly broader idea than you what is meant by "System". Just calling the annotation "ExtraData" seems not descriptive enough to me, but I suppose it could reduce confusion somewhat.

Reply:
It would have been nice to just have `@RegisterExtraData` on both the class and the methods, but there is additional information required for the method annotation, so different annotation classes (and thus different names) are needed.

Reply:
> ExtraDataSystem that isn't a ComponentSystem

oh interesting. I had indeed not considered that scenario. To me when I think about other variants of "x is a `ComponentSystem`" I think of "x is an `UpdateSubscriberSystem`" - as in explicit implementing `System` class types only

If you think the current approach makes sense then I'm good with it :-)